### PR TITLE
Add some missing translator comments and check for missing translator comments in future

### PR DESCRIPTION
**Pull request by jcsteh**
### Link to issue number:
None specifically, but inspired by an omition in #7258.

### Summary of the issue:
All translatable strings should have translator comments explaining them to translators. However, this is sometimes forgotten and this is easy to miss in review.

### Description of how this pull request fixes the issue:
1. Add a few missing translator comments inspired by [reports from translators](https://www.freelists.org/post/nvda-translations/Missing-comment-for-translators-was-About-a-string-for-NVDA-20173).
2. Add some code (`checkPot`) to check for missing translator comments in the translation template (pot). For now, we have a set of expected failures, since fixing all existing 87 messages that don't have translator comments is going to take some work. However, we want to fail for any new messages that get introduced without comments.
3. Run `checkPot` as part of `scons tests`. This means tests (and thus builds) will fail so we learn about these problems early.
4. `checkPot` can also be run alone with `scons checkPot`.
5. Now that `scons tests` runs `checkPot` (which depends on `pot`), don't explicitly run `scons pot` on AppVeyor.

### Testing performed:
1. Tested that unexpected errors cause `checkPot` to fail.
2. Tested that expected errors don't cause `checkPot` to fail, but do get counted in the summary.
3. Added a message that already has a translator comment to the `EXPECTED_MESSAGES_WITHOUT_COMMENTS` set. Tested that this causes an unexpected success and that `checkPot` fails as a result.
4. Tested that `scons tests` runs `checkPot`.
5. Tested that `scons tests unitTests=test_cursorManager` does *not* run `checkPot` (since explicit tests were specified).
6. Tested that an unexpected error causes an AppVeyor build to fail.

### Known issues with pull request:
None.

### Change log entry:
Changes for developers:

```
- "scons tests" now checks that translatable strings have translator comments. You can also run this alone with "scons checkPot". (#7492)
```

### CODE REVIEW

**Comment by jcsteh** [comment]
@MichaelDCurran ended up fixing the string in #7258 in master 890528bd. I've modified this PR accordingly.

File: tests/checkPot.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+			if line.startswith("#. Translators: "):
+				# Example: "#. Translators: a message reported in the SetRowHeader script for Microsoft Word."
+				hasComment = True
+				continue
+			if line.startswith("#: "):
+				# Example: "#: NVDAObjects\window\winword.py:1322"
+				# Strip the "#: " prefix (3 chars).
+				sourceLines.append(line[3:])
+				continue
+			if line.startswith('msgctxt "'):
```
Comment (feerrenrut) [inline tests/checkPot.py]: It's probably assumed knowledge for this, but what are: `msgctxt`, `msgid`, `msgstr`. Are these concepts from some other library module? Either a brief explanation of what they are or a pointer to where to find out about them could be handy for someone who comes to maintain this without that background knowledge.

File: tests/checkPot.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+"""Check a translation template (pot) for strings without translator comments.
+"""
+
+import sys
+
+# Existing messages that we know don't have translator comments yet.
+# Ideally, all of these should get translator comments,
+# but this is not realistic right now.
+# A message should be removed from here once a translator comment is added for it.
```
Comment (feerrenrut) [inline tests/checkPot.py]: Found out it does, could you update this comment to specify that it does?

**Review by feerrenrut — COMMENTED** [review]
Looks good overall, just few comments.

**Comment by jcsteh** [comment]
@feerrenrut, thanks; I think I've addressed your review comments... with
code comments. :)

**Review by feerrenrut — APPROVED** [review]

**Pull request merged**